lib/Stats.js

```diff
@@ -439,6 +449,23 @@ Stats.jsonToString = function jsonToString(obj, useColors) {
 			.toPrecision(3) + " " + abbreviations[index];
 	}
 
+	function isHugeBundle(assetSize) {
+		var sizeLimit = obj.performance.maxAssetSize;
+		return sizeLimit < assetSize;
+	}
+
+	function getAssetColor(asset, defaultColor) {
+		var jsRegex = /\.js($|\?)/i;
+
+		if("performance" in obj) {
+			if(isHugeBundle(asset.size) && jsRegex.test(asset.name)) {
```

Comment:
such computations should be done in a plugin and results stored in a property on the compilation.
Stats should only display things and don't calculate them. This also removes the need for `compilation.compiler.options`